### PR TITLE
Alerting: New search UI – Part 1

### PR DESCRIPTION
**Pull request by gillesdemey**
This PR sets up the new search UI by;

* Moving the files around to `rules/Filter/*`
* Renaming `RulesFilter` to `RulesFilter.v1`
* Adding `RulesFilter.v2`
* Creating a wrapper component `RulesFilter` that conditionally render a different component

Additionally refactors `HoverCard` to `PopupCard` since we should also be able to invoke it on click interaction.

The click interaction is currently also not great - interacting with the dropdown components inside the card will dismiss it.

The new search UI is completely non-functional and subject to change.

image

### CODE REVIEW

File: public/app/features/alerting/unified/components/rule-list/RuleList.v1.tsx [hunk 1]
```diff
@@ -105,28 +103,24 @@ const RuleList = withErrorBoundary(
...
-                  >
-                    {expandAll ? 'Collapse all' : 'Expand all'}
-                  </Button>
-                )}
-                <RuleStats namespaces={filteredNamespaces} />
-              </div>
-            </div>
-          </>
-        )}
+        <Stack direction="column">
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/components/rule-list/RuleList.v1.tsx:106]: Why do we need this stack?

File: public/app/features/alerting/unified/components/rules/Filter/RulesFilter.v2.tsx [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    );
+  }, [activeTab, styles.content, styles.fixTabsMargin]);
+
+  return (
+    <Stack direction="column" gap={0}>
+      <Label>
+        <Trans i18nKey="common.search">Search</Trans>
+      </Label>
+      <Stack direction="row">
+        <Input prefix={filterOptions} />
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/components/rules/Filter/RulesFilter.v2.tsx:74]: what's the reason using an input here with the prefix?

**Review by soniaAguilarPeiron — APPROVED** [review]
Code changes LGTM!🚀
Added a non blocking comment.

File: public/app/features/alerting/unified/components/rule-list/RuleList.v1.tsx (hunk 1 above)
Comment (gillesdemey) [reply public/app/features/alerting/unified/components/rule-list/RuleList.v1.tsx:106]: Oh probably a left-over from refactoring something locally? We certainly don't need it here – I'll revert this bit.

**Pull request merged**